doc/sphinx/language/core/definitions.rst

```diff
@@ -151,21 +164,23 @@ The basic assertion command is:
       | Proposition
       | Property
 
-   After the statement is asserted, Rocq needs a proof. Once a proof of
-   :n:`@type` under the assumptions represented by :n:`@binder`\s is given and
-   validated, the proof is generalized into a proof of :n:`forall {* @binder }, @type` and
-   the theorem is bound to the name :n:`@ident` in the global environment.
+   In its most basic form :g:`@thm_token @ident : @type`, the command
+   interprets :n:`@type` as a statement and waits for a
+   proof. Once a proof is given and validated, a theorem bound to
+   :n:`@ident` and stating :n:`@type` is added to the global
+   environment.
 
-   These commands accept the :attr:`program` attribute.  See :ref:`program_lemma`.
+   The form :n:`@thm_token @ident {* @binder } : @type := @term` is
+   equivalent to :n:`@thm_token @ident : forall {* @binder }, @type`
+   and the proof starts in the context :n:`{* @binder }`.
 
    Forms using the :n:`with` clause are useful for theorems that are proved by simultaneous induction
-   over a mutually inductive assumption, or that assert mutually dependent
-   statements in some mutual coinductive type. It is equivalent to
+   over a mutually inductive assumption, or that assert mutually dependent coinductive
+   statements. It is equivalent to
    :cmd:`Fixpoint` or :cmd:`CoFixpoint` but using tactics to build the proof of
    the statements (or the :term:`body` of the specification, depending on the point of
    view). The inductive or coinductive types on which the induction or
-   coinduction has to be done is assumed to be unambiguous and is guessed by
-   the system.
+   coinduction has to be done is guessed by the system.
```

Comment:
-> The command guesses the inductive or coinductive types on which the induction or
   coinduction has to be done.

Clearer without the passive voice